enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -493,27 +497,30 @@ func NewSchedulerServerWithOptions(env environment.Env, options *Options) (*Sche
 	}
 
 	s := &SchedulerServer{
-		env:                          env,
-		pools:                        make(map[nodePoolKey]*nodePool),
-		rdb:                          env.GetRemoteExecutionRedisClient(),
-		taskRouter:                   taskRouter,
-		schedulerClientCache:         newSchedulerClientCache(),
-		shuttingDown:                 shuttingDown,
-		enableUserOwnedExecutors:     enableUserOwnedExecutors,
-		requireExecutorAuthorization: requireExecutorAuthorization,
-		ownHostPort:                  fmt.Sprintf("%s:%d", ownHostname, ownPort),
+		env:                           env,
+		pools:                         make(map[nodePoolKey]*nodePool),
+		rdb:                           env.GetRemoteExecutionRedisClient(),
+		taskRouter:                    taskRouter,
+		schedulerClientCache:          newSchedulerClientCache(),
+		shuttingDown:                  shuttingDown,
+		enableUserOwnedExecutors:      enableUserOwnedExecutors,
+		forceUserOwnedDarwinExecutors: forceUserOwnedDarwinExecutors,
+		requireExecutorAuthorization:  requireExecutorAuthorization,
+		ownHostPort:                   fmt.Sprintf("%s:%d", ownHostname, ownPort),
 	}
 	return s, nil
 }
 
-func (s *SchedulerServer) GetGroupIDAndDefaultPoolForUser(ctx context.Context) (string, string, error) {
+func (s *SchedulerServer) GetGroupIDAndDefaultPoolForUser(ctx context.Context, os string) (string, string, error) {
 	defaultPool := s.env.GetConfigurator().GetRemoteExecutionConfig().DefaultPoolName
 	if !s.enableUserOwnedExecutors {
 		return "", defaultPool, nil
 	}
-
 	user, err := perms.AuthenticatedUser(ctx, s.env)
 	if err != nil {
+		if s.forceUserOwnedDarwinExecutors && os == "darwin" {
```

Comment:
this check should go inside the following if statement

Reply:
Done